packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2824,4 +2826,21 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         this.state.isDeclareContext = oldIsDeclareContext;
       }
     }
+
+    parseMethod(...args: any[]) {
+      const method = super.parseMethod(...args);
+      if ((method: any).abstract) {
+        const hasBody = this.hasPlugin("estree")
+          ? !!method.value.body
+          : !!method.body;
+        if (hasBody) {
+          this.raise(
+            method.start,
+            TSErrors.AbstractMethodHasImplementation,
+            (method: any).key.name,
```

Comment:
The method key is not always Identifier, and that's why Flow complains here.

```ts
abstract class C {
  abstract [M]();
}
```

I think we can just throw
```
An abstract method cannot have an implementation.
```

Reply:
Is this correct error message to be used here? For example TS never throws on 
`abstract class C {  abstract a(): string; }`, but throws on `abstract class C {  abstract a(); }` because it's lack return type.

Regarding the example above. TS throws

`A computed property name in a method overload must refer to an expression whose type is a literal type or a 'unique symbol' type.ts(1168`

`'[M]', which lacks return-type annotation, implicitly has an 'any' return type.ts(7010)`

Reply:
If you disable the (opt-in, but enabled by default in the playground) `noImplicitAny` option, that code doesn't generate any error: https://www.typescriptlang.org/play?noImplicitAny=false#code/MYewdgzgLgBAsjAvDARAawKYE8UwIYQyiRQDcAsAFBV4BG0ATnsLMADYGEDCMA3lTHz0oTFjADacALoAKAJQVKAXypA

Reply:
Sorry for the confusion here, I was meant to say that an abstract method can be computed, so that example is valid. And the TS error can be suppressed by specifying `M`. To be clear

```ts
abstract class C {
  abstract [M]() {}
}
```
will throw.

I don't think we have to extract computed key here, since it can be very verbose.

Reply:
> I don't think we have to extract computed key here, since it can be very verbose.

I ended up doing it since I noticed it's exactly what TS does.